### DISCUSSION TITLE
How do I compile mandelbrot.c for cc64? Driver issue

### DISCUSSION THREAD

**Discussion by gerroon** [cc65/cc65#2393] [Q&A] [answered]
Hi
I wanted to run mandelbrot.c but I am hitting an error when I compile and run it in Vice Windows, it seems to like it is missing a driver :( I tested with source compiled cc65 on Linux and the release binaries for Windows x65

![image](https://github.com/cc65/cc65/assets/8519469/3860efeb-0119-4908-a9de-4165a669ac2e)

**Comment by mrdudz** [comment, marked as answer]
You need the (a) TGI driver. In the samples directory, use "make disk" to build a d64 with everything on it that is needed

**Reply by gerroon** [reply]
Hi, thanks for the reply, but I do not understand how that is done with the given cc65 tools :( I found this TGI zip download on the web, but not sure how I use these to make the disk you mentioned.

![image](https://github.com/cc65/cc65/assets/8519469/905c2ca2-3053-4140-bbb6-43dc62f00463)

**Comment by mrdudz** [comment]
Everything is included in the cc65 distribution - no need to download any extra packages

**Reply by gerroon** [reply]
Thanks, then I really do not know how to achieve/create what I need to run the mandelbrot.c on C64. I do not see a make disk binary in the release for instance. There is no info on the net on how to use the TGI driver with cc65 either or I could not find what that is.

**Reply by polluks** [reply]
for your information
https://github.com/cc65/cc65/blob/5c3ff714aebfdf3e5c83c180248d7260add1afbe/samples/readme.txt#L29
https://cc65.github.io/doc/tgi.html

**Comment by gerroon** [comment]
I could not figure out the way @mrdudz  was recommending, which I believe making a disk with the driver. So here is how to make it work bypassing his solution.


I am still interested in figuring out his solution, so I wouldbe happy if someone offers a clear instruction to achieve it, Until then this is what worked for me.


```
Uncomment the comacro and add the lines below. 
// #if DYN_DRV
    // /* Load the graphics driver */
    // cprintf ("initializing... mompls\r\n");
    // tgi_load_driver (tgi_stddrv);
// #else
    // /* Install the graphics driver */
    // tgi_install (tgi_static_stddrv);
// #endif
	
  if (argc==1)
    tgi_install(tgi_static_stddrv);
  else
    tgi_load_driver(argv[1]);

  if (tgi_geterror()!=TGI_ERR_OK)
    return 1;

```

**Reply by polluks** [reply]
You can execute `CFLAGS=-DDYN_DRV=0 make clean mandelbrot` to turn off the dynamic driver.
Now you don't need a d64 because it's just one file.
